src/llvm/di_builder.cr

```diff
@@ -95,7 +95,11 @@ struct LLVM::DIBuilder
   end
 
   def insert_declare_at_end(storage, var_info, expr, dl : LibLLVM::MetadataRef, block)
-    LibLLVM.di_builder_insert_declare_at_end(self, storage, var_info, expr, dl, block)
+    {% if LibLLVM::IS_LT_190 %}
+      LibLLVM.di_builder_insert_declare_at_end(self, storage, var_info, expr, dl, block)
+    {% else %}
+      LibLLVM.di_builder_insert_declare_record_at_end(self, storage, var_info, expr, dl, block)
+    {% end %}
```

Comment:
Oh my, we're returning a raw `Void*` from LLVM here, which may be a `LibLLVM::ValueRef` or a `LibLLVM::DbgRecordRef` now :disappointed:

Reply:
There's only a single call to this method in the compiler, and the return value is ignored. Maybe we should tack on a type restriction to `Nil`? :shrug:

Reply:
For the crystal compiler we may not care, but the method is public...

Reply:
Or we do the sensible thing of wrapping those pointers into an actual `LLVM::Metadata` or `LLVM::DebugRecord` struct

Reply:
Also `LLVM::DIBuilder` is marked as `@[Experimental]`, so we are free to make breaking changes here, and probably not in a hurry to do so